File: java/counting_sort.java
import java.util.*;

class counting_sort{

	public static void cs(int arr[],int n,int k){
		int c[]=new int[k+1];
		int b[]=new int[n];
		for(int i=0;i<=k;i++){
			c[i]=0;
		}

		for(int j=0;j<n;j++){
			c[arr[j]]=c[arr[j]]+1;
		}
		for(int i=1;i<=k;i++){
			c[i]=c[i]+c[i-1];
		}
		for(int j=0;j<n;j++){
			b[c[arr[j]]-1]=arr[j];
			c[arr[j]]=c[arr[j]]-1;
		}

		System.out.println("Elements after counting sort:");
		for(int i=0;i<n;i++){
			System.out.print(b[i]+"\t");
		}
	}

	public static void main(String args[]){
		int n,max=0;
		int arr[]=new int[100];

		System.out.println("Enter the value  of n:");
		Scanner sc=new Scanner(System.in);

		n=sc.nextInt();
		System.out.println("Enter "+n+" elements:");
		for(int i=0;i<n;i++){
			arr[i]=sc.nextInt();
			if(max<=arr[i]){
				max=arr[i];
			}
		}
		
		cs(arr,n,max);
		
	}
}

File: python/counting_sort.py
def counting_sort(arr,n,k):
	b=[]
	c=[]
	for i in range(k+1):
		c.append(0)
	for j in range(n):
		c[arr[j]]+=1
		b.append(0)
		
	for i in range(1,k+1):
		c[i]+=c[i-1]
	for j in range(n):
		b[c[arr[j]]-1]=arr[j]
		c[arr[j]]-=1
	print("Elements after counting sort:")
	print(b)

print("Enter the value of n:")
n=int(input())
li=[]

print("Enter "+str(n)+" elements:")
for i in range(n):
	li.append(int(input()))
counting_sort(li,n,max(li))
